anilistUpdater/main.lua

```diff
@@ -221,19 +219,17 @@ function check_progress()
     if percent_pos >= UPDATE_PERCENTAGE then
         update_anilist("update")
         triggered = true
-        if progress_timer then
-            progress_timer:stop()
-        end
+        progress_timer:stop()
         return
     end
 end
 
 -- Handle pause/unpause events to control the timer
 function on_pause_change(name, value)
-    if progress_timer then
-        if value then
-            progress_timer:stop()
-        else
+    if value then
+        progress_timer:stop()
```

Comment:
You removed the progress_timer check so it automatically loads the file into a nil error. Should probably move the `progress_timer` init to above the `on_pause_change` function and do a ` ~= nil` check

Otherwise everything else looks good